plugins/nf-sqldb/src/main/nextflow/sql/InsertHandler.groovy

```diff
@@ -157,7 +165,19 @@ class InsertHandler implements Closeable {
             for(int i=0; i<columns.size(); i++ ) {
                 final col = columns[i]
                 final value = record.get(col)
-                stm.setObject(i+1, value)
+                try {
+                    stm.setObject(i+1, value)
+                }
+                catch(UnsupportedOperationException e) {
+                    log.debug "setObject is not supported by this driver (likely Databricks), trying setString fallback: ${e.message}"
+                    // Fallback to setString for basic types
+                    try {
+                        stm.setString(i+1, value?.toString())
+                    }
+                    catch(Exception fallbackError) {
+                        throw new RuntimeException("Failed to set parameter ${i+1} with value '${value}': both setObject and setString failed", fallbackError)
+                    }
+                }
```

Comment:
Since it's a recurrent pattern, it may be worth to turn into a small function e.g 

```
void setValue(Statement stm, int index, Object value) {
  try {
    stm.setObject(i, value)
  }
  catch( UnsupportedOperationException e ) {
     stm.setString(i, value?.toString())
  } 
}
```

and let it fails in other cases